### ISSUE TITLE
cmd/go: go test -json has inconsistent Time precision

### ISSUE THREAD

**Issue by rittneje** [golang/go#44588]
### What version of Go are you using (`go version`)?

<pre>
$ go version
go version go1.14.13 darwin/amd64
</pre>

### Does this issue reproduce with the latest release?

Yes (go1.16)

### What did you do?

Ran `go test -json` on a package, and attempted to sort the resulting JSON via jq using the `Package` and `Time` fields.

### What did you expect to see?

I expected jq to sort the results correctly.

### What did you see instead?

I noticed that the decimal precision of the `Time` field is inconsistent. My guess is that it is chopping off trailing zeros. However, this means that the resulting timestamps cannot be sorted properly. For example, we might get the following two entries:

> {"Time":"2021-02-24T14:14:25.8589Z","Action":"output","Package":"my-package","Test":"TestFoo","Output":"=== RUN TestFoo\n"}
> {"Time":"2021-02-24T14:14:25.8589131Z","Action":"output","Package":"my-package","Test":"TestFoo","Output":" --- PASS: TestFoo (0.00s)\n"}

Since 'Z' > '1', this causes the "PASS" line to get sorted before the "RUN" line. This is especially a problem when dealing with older (< 1.6) versions of jq that do not perform a stable sort.

**Comment by ChandanChainani** [comment]
@rittneje please can you check with go version `1.19` and `1.20`, if failed please share stes and cmd.

**Comment by seankhliao** [comment]
This is working as intended for time.Time.

**Comment by rittneje** [comment]
@seankhliao I disagree with that assessment. time.Time may be abiding by the format string, but that doesn’t change the fact that go test should pass one that won’t have variable precision.
